kanidm_unix_int/src/cache.rs

```diff
@@ -264,17 +266,36 @@ impl CacheLayer {
             .map_err(|e| {
                 error!("time conversion error - ex_time less than epoch? {:?}", e);
             })?;
+        // WIP #392: check user/default shell
+        // check if the provided `shell` exists on the system.
+        let mut new_token = token.clone();
```

Comment:
You don't need this clone. Change this line:

```
async fn set_cache_usertoken(&self, token: UnixUserToken) -> Result<(), ()> {
```

to

```
async fn set_cache_usertoken(&self, mut token: UnixUserToken) -> Result<(), ()> {
```

Reply:
I am trying to get this running. It is a `&UnixUserToken` 

`async fn set_cache_usertoken(&self, mut token: &UnixUserToken) -> Result<(), ()> {`

bit if it is a `&` ref, I cannot change the value. Therefore, I cloned it ... 

If I try to do it via 
`async fn set_cache_usertoken(&self, mut token: UnixUserToken) -> Result<(), ()> {`

I get some other problems ... (sorry this is not very specific)